### PR TITLE
feat: add method for `janus-cli package metadata` ; generate updates to package.json for all plugins listed in plugins/plugin-metadata.json

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

feat: add method for `janus-cli package metadata` [RHIDP-1502](https://issues.redhat.com//browse/RHIDP-1502) ;
generate updates to package.json for all plugins listed in plugins/plugin-metadata.json

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?

* check out PR
* run stuff like thiis:

```
./packages/cli/bin/janus-cli package metadata --help
./packages/cli/bin/janus-cli package metadata --dir plugins/acr --keywords "lifecycle:foo,support:baz,feature:stuff"
cd plugins/argocd; ../../packages/cli/bin/janus-cli package metadata --author foo --license bar --homepage http://bazbar --bugs https://woot
```

Or use the script wrapper to run against a list of plugins:
```
cat ./plugins/plugin-metadata.json
./plugins/plugin-metadata.sh
```

Followup tasks would be to contribute this script / app to upstream community-plugins repo, so we can run it against existing/future plugins and incorporate it into lifecycle automation.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
Wrote >300 lines of code:

```
git diff --shortstat 59276dfb 11f3b5a2 -- . ':!*yarn.lock'
 8 files changed, 335 insertions(+), 6 deletions(-)
```
... to generate 268 lines of changed code (434-166) ...

```
$➔ git diff --shortstat 11f3b5a2 9e6dfc6a  -- . ':!*yarn.lock'
 43 files changed, 434 insertions(+), 166 deletions(-)
```

Yeah that seems efficient :lolsob:

**Comment by nickboldt** [comment]
Not sure if I need this to help resolve the git configlocal module:

```
$➔ gd .
diff --git a/packages/cli/tsconfig.json b/packages/cli/tsconfig.json
index 1119aafd..a9471368 100644
--- a/packages/cli/tsconfig.json
+++ b/packages/cli/tsconfig.json
@@ -1,10 +1,14 @@
 {
   "extends": "@backstage/cli/config/tsconfig.json",
-  "include": ["src"],
+  "include": ["src", "types.d.ts"],
   "exclude": ["node_modules", "templates"],
   "compilerOptions": {
     "outDir": "../../dist-types/packages/cli",
     "rootDir": ".",
-    "useUnknownInCatchVariables": false
+    "useUnknownInCatchVariables": false,
+    "moduleResolution": "node",
+    "module": "commonjs",
+    "resolveJsonModule": true,
+    "isolatedModules": true
   }
 }
```

File: plugins/orchestrator-backend/package.json [hunk 1]
```diff
@@ -92,5 +99,10 @@
     "@types/express": "4.17.20",
     "@types/fs-extra": "11.0.1",
     "@types/json-schema": "7.0.12"
-  }
+  },
+  "maintainers": [
+    "@janus-idp/maintainers-plugins",
+    "@janus-idp/orchestrator-codeowners"
+  ],
+  "author": "'The"
```
Comment (caponetto) [inline plugins/orchestrator-backend/package.json:107]:
```suggestion
  "author": "Red Hat"
```
?
There are other packages with the same value.

Comment (nickboldt) [reply plugins/orchestrator-backend/package.json:107]: odd. I'll look into what the heck happened there. It should definitely be `Red Hat` unless the plugin is owned/contrib'd by someone outside RH.

Comment (nickboldt) [reply plugins/orchestrator-backend/package.json]: fixed in https://github.com/janus-idp/backstage-plugins/pull/1627/commits/ae211d2c40c5aca045c335b1cee5e579f1ec55d2#diff-e3fb4fd157240487ad11e13af78c9bf58c055218b2f3ce9dbd897201fa585f83R12 ... turns out there's value to all the bash linting reccos I was ignoring :D

Comment (nickboldt) [reply plugins/orchestrator-backend/package.json]: @caponetto  Would you prefer if these 4 plugins were authored by RH ?

```
$➔ for d in plugins/orchestrator*/package.json; do echo; echo $d; jq .author $d; done

plugins/orchestrator-backend/package.json
"The Backstage Community"

plugins/orchestrator-common/package.json
"The Backstage Community"

plugins/orchestrator/package.json
"The Backstage Community"

plugins/orchestrator-swf-editor-envelope/package.json
"The Backstage Community"
```

Comment (caponetto) [reply plugins/orchestrator-backend/package.json]: @pkliczewski do you mind answering these questions, please? ^

**Comment by pkliczewski** [comment]
@caponetto @jkilzi @batzionb please take a look

File: plugins/feedback-backend/package.json [hunk 2]
```diff
@@ -72,11 +73,22 @@
...
   ],
-  "homepage": "https://janus-idp.io/",
-  "bugs": "https://github.com/janus-idp/backstage-plugins/issues"
+  "homepage": "https://red.ht/rhdh",
+  "bugs": "https://github.com/janus-idp/backstage-plugins/issues",
+  "maintainers": [
+    "@janus-idp/maintainers-plugins",
+    "@janus-idp/devex-uxe"
+  ],
+  "author": "The Backstage Community"
```
Comment (yashoswalyo) [inline plugins/feedback-backend/package.json:93]: @riginoommen @deshmukhmayur what should be there in author field?

File: packages/cli/src/types.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 The Backstage Authors
+ * Copyright 2020-2024 The Janus Authors
```
[the remark this answers is not in the document]
Comment (nickboldt) [reply packages/cli/src/types.d.ts:2]: it's in our fork, so yes. But sure, we can leave this alone with the outdated copyright and wrong owner :D

**Comment by nickboldt** [comment]
replaced by #1753

**Pull request closed without merging**